lib/kitchen/provisioner/salt_solo.rb

```diff
@@ -238,23 +240,30 @@ def init_command
         cmd
       end
 
+      def os_join(*args)
+        if windows_os?
+           File.join(*args).tr('/', '\\')
+        else
+           File.join(*args)
+        end
+      end
+
       def salt_command
         salt_version = config[:salt_version]
 
         cmd = ''
         if windows_os?
           salt_call = "c:\\salt\\salt-call.bat"
-          salt_config_path = config[:salt_config].tr('/', '\\')
-          cmd << "(get-content #{File.join(config[:root_path], salt_config_path, 'minion').tr('/', '\\')}) -replace '\\$env:TEMP', $env:TEMP | set-content #{File.join(config[:root_path], salt_config_path, 'minion').tr('/', '\\')} ;"
-          cmd << sudo("#{salt_call} --state-output=changes --config-dir=#{File.join(config[:root_path], salt_config_path).tr('/', '\\')} --local state.highstate")
+          salt_config_path = config[:salt_config]
+          cmd << "(get-content #{os_join(config[:root_path], salt_config_path, 'minion')}) -replace '\\$env:TEMP', $env:TEMP | set-content #{os_join(config[:root_path], salt_config_path, 'minion')} ;"
         else
           # install/update dependencies
           cmd << sudo("chmod +x #{config[:root_path]}/*.sh;")
           cmd << sudo("#{config[:root_path]}/dependencies.sh;")
           salt_config_path = config[:salt_config]
           salt_call = 'salt-call'
-          cmd << sudo("#{salt_call} --state-output=changes --config-dir=#{File.join(config[:root_path], salt_config_path)} --local state.highstate")
         end
+        cmd << sudo("#{salt_call} --state-output=changes --config-dir=#{os_join(config[:root_path], salt_config_path)} --local state.highstate")
```

Comment:
neat, also if we add `local: true` to the minion.erb, we can remove the `--local` here, so that if people want to hook the test minion up to a master, they can, and then provide their own minion.erb file that removes local: true.